x/pairing/keeper/pairing.go

```diff
@@ -291,13 +291,10 @@ func (k Keeper) calculatePairingForClient(ctx sdk.Context, providers []epochstor
 
 	if spec.ProvidersTypes == spectypes.Spec_dynamic {
 		// calculates a hash and randomly chooses the providers
-
-		validProviders = k.returnSubsetOfProvidersByStake(ctx, developerAddress, providers, providersToPair, epochStartBlock, chainID, epochHash)
-	} else {
-		validProviders = k.returnSubsetOfProvidersByHighestStake(ctx, providers, providersToPair)
+		providers = k.returnSubsetOfProvidersByStake(ctx, developerAddress, providers, providersToPair, epochStartBlock, chainID, epochHash)
```

Comment:
it is not usual to rewrite a parameter received by the method,

instead 
if  spec.ProvidersTypes == spectypes.Spec_dynamic {
   return k.returnSubsetOfProvidersByStake(ctx, developerAddress, providers, providersToPair, epochStartBlock, chainID, epochHash),nil
}
return providers,nil

Reply:
Fixed 23d804640a940ee2e5215adab9e2588813086266